### PR TITLE
Combine similar if logics in core

### PR DESCRIPTION
**Pull request by hussein-awala**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by Lee-W — APPROVED** [review]

File: airflow/models/baseoperator.py [hunk 1]
```diff
@@ -1107,9 +1107,11 @@ def dag(self, dag: DAG | None):
...
         if self.__from_mapped:
             pass  # Don't add to DAG -- the mapped task takes the place.
-        elif self.task_id not in dag.task_dict:
-            dag.add_task(self)
-        elif self.task_id in dag.task_dict and dag.task_dict[self.task_id] is not self:
+        elif (
+            self.task_id not in dag.task_dict
+            or self.task_id in dag.task_dict
+            and dag.task_dict[self.task_id] is not self
+        ):
```
Comment (uranusjr) [inline airflow/models/baseoperator.py]:
```suggestion
        elif (
            self.task_id not in dag.task_dict
            or dag.task_dict[self.task_id] is not self
        ):
```

or just

```suggestion
        elif dag.task_dict.get(self.task_id) is not self:
```

File: airflow/ti_deps/deps/trigger_rule_dep.py [hunk 2]
```diff
@@ -367,20 +367,7 @@ def _iter_upstream_conditions() -> Iterator[ColumnOperators]:
...
-                yield self._failing_status(
-                    reason=(
-                        f"Task's trigger rule '{trigger_rule}' requires all upstream tasks to have "
-                        f"succeeded or been skipped, but found {num_failures} non-success(es). "
-                        f"upstream_states={upstream_states}, "
-                        f"upstream_task_ids={task.upstream_task_ids}"
-                    )
-                )
-        elif trigger_rule == TR.NONE_FAILED_MIN_ONE_SUCCESS:
+        elif trigger_rule in (TR.NONE_FAILED, TR.NONE_FAILED_MIN_ONE_SUCCESS):
```
Comment (uranusjr) [inline airflow/ti_deps/deps/trigger_rule_dep.py]: Not sure I like using `in` here; an `or` combination of `==` would be better.

File: airflow/models/baseoperator.py (hunk 1 above)
Comment (hussein-awala) [reply airflow/models/baseoperator.py]: Indeed, the second suggestion should do the job.

**Review by potiuk — APPROVED** [review]

**Review by uranusjr — APPROVED** [review]

**Comment by potiuk** [comment]
conflicts, I am afraid

**Pull request merged**